src/modules/users/users.controller.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  HttpStatus,
+  Post,
+} from '@nestjs/common';
+import { UsersService } from './users.service';
+import { User } from 'src/common/entities/user.entity';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+@ApiTags('users')
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+  @Get()
+  @ApiOperation({ summary: 'Get all users' })
+  @ApiResponse({ status: 200, description: 'Return all users.', type: [User] })
+  async getUsers(): Promise<User[]> {
+    try {
+      return await this.usersService.findAllUsers();
+    } catch (error) {
+      throw new HttpException(
+        'Internal Server Error',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new user' })
+  @ApiResponse({
+    status: 201,
+    description: 'The user has been successfully created.',
+    type: User,
+  })
+  async addUser(
+    @Body('name') name: string,
+    @Body('email') email: string,
+  ): Promise<User> {
+    try {
```

Comment:
The try-catch block in the addUser method might be unnecessary, as NestJS has built-in error handling. Perhaps it's better to use NestJS's global exception handling for cleaner code.

Reply:
agree with comment